x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -270,6 +271,10 @@ public enum DataType implements Writeable {
      * Nanosecond precision date, stored as a 64-bit signed number.
      */
     DATE_NANOS(builder().esType("date_nanos").estimatedSize(Long.BYTES).docValues().supportedOnAllNodes()),
+    /**
+     * Represents a half-inclusive range between two dates.
```

Comment:
Is it half-inclusive? I saw a comment in the RangeFieldMapper that indicates this is fully inclusive.